rosetta-source/src/main/rosetta/legaldocumentation-csa-type.rosetta

```diff
@@ -1,8 +1,182 @@
+    floatingRate CsaFloatingRate(0..1) <"Specifies the floating interest rate to be used.">
+    inBaseCurrency boolean(1..1) <"Specifies whether valuation transfers should be converted to base currency equivalent.">
+    compoundingType CompoundingTypeEnum(0..1) <"Specifies the type of compounding to be applied (None, Business, Calendar).">
+    compoundingBusinessCenter BusinessCenterEnum(0..*) <"Specifies the applicable business centers for compounding.">
+    dayCountFraction DayCountFractionEnum(1..1) <"Specifies the day count fraction to use for that currency.">
+    rounding Rounding(0..1) <"Specifies the rounding rules for settling in that currency.">
+    roundingFrequency RoundingFrequencyEnum(0..1) <"Specifies when/how often is rounding applied?">
+    withholdingTaxRate number(0..1) <"Specifies the withholding tax rate if a withholding tax is applicable.">
+
+    condition  InterestRate: required choice fixedRate, floatingRate
+    condition DCF:  dayCountFraction = DayCountFractionEnum->ACT_360 or dayCountFraction = DayCountFractionEnum->ACT_365_FIXED
+    condition CompoundingBC1:  if compoundingType = CompoundingTypeEnum->Business then compoundingBusinessCenter exists
+    condition CompoundingBC2:  if compoundingType <> CompoundingTypeEnum->Business then compoundingBusinessCenter is absent
+
+/*  Previous version of CsaFloatingRate, kept for review purposes; delete before publishing 
+type CsaFloatingRate: <"Represents the parameters needed to calculate the floating rate paid on collateral holdings.">
+    index string(1..1) <"Specifies the floating rate index to use for that currency.">
+    spread number(0..1) <"Specifies any fixed spread added to the floating rate.">
+    capRate number(0..1) <"Specifies any cap rate applicable;  if a cap is applicable, the rate used is the maximum of the cap rate and the floating rate.">
+    floorRate number(0..1) <"Specifies any floor rate applicable; if provided, the rate used is the minimum of the floor rate and the floating rate.">
+    negativeInterest boolean (1..1) <"Specifies how negative rates should be applied.  If rates go negative, should the payment be reversed (true) or zeroed out (false)?">
+    compressibleSpread boolean( 1..1) <"Specifies how spreads should be applied in a low/negative rate environment.  If true, spread is applied only if rate is positive.">
+*/
+
+
+type CsaFloatingRate extends FloatingRateBase: <"Represents the parameters needed to calculate the floating rate paid on collateral holdings.">
+    negativeInterest boolean (1..1) <"Specifies how negative rates should be applied.  If rates go negative, should the payment be reversed (true) or zeroed out (false)?">
+    compressibleSpread boolean( 1..1) <"Specifies how spreads should be applied in a low/negative rate environment.  If true, spread is applied only if rate is positive.">
+
+    /*  Following is a table that shows how to apply the negativeInterest and compressibleSpread indicators
```

Comment:
In theory this kind of comment should be placed in the relevant model description. However those model descriptions do not allow formatting (such as a table), so fine to keep it as such (for now).